### DISCUSSION TITLE
Game is very dark

### DISCUSSION THREAD

**Discussion by madig** [Try/OpenGothic#459] [General]
Hi! I stumbled over this project recently, exciting.

I'm currently playing Chronicles of Myrtana and re-discovered how dark Gothic is, as in, I have to turn up the gamma because I can't see in broad daylight. In OpenGothic, the gamma sliders don't work...

I noticed that the raytracing discussion thread was talking about how the game was made for a "wrong lighting" model and doing it correctly will make it appear very dark. Is there anything that can be done? The DX11 mod for the original had the same problem, and "HDR" made the game look... off.

Is there anything that can be done?

**Comment by Try** [comment]
Hi, @madig !

> In OpenGothic, the gamma sliders don't work...

Yeah, it's not implemented - I've put it into backlog.

> "wrong lighting" model and doing it correctly will make it appear very dark. Is there anything that can be done?

not necessarily dark, just "wrong". Also I'm working on lighting model in slow phase, improving thing bit-by-bit.
State for today:
* Lighting is HDR only. Screenshoot on main page is HDR. 
Also I've noted that HDR in DX11-mod looks off, maybe it's not finished/broken - don't know for sure.
* Sky/Sun/Moon/Stars do have physical-based Illuminance,  
* Exposure is single `float` and based on daytime - not great, not terrible
* Local light are using modified filament-curve for falloff 

Magic/VFX and local light are special case still. With magic - it's common problem in current-gen engines. No one knows how many LUX/watts in paladins holly-arrow. And there is good values, as shown here:
![Screenshoot](https://knarkowicz.files.wordpress.com/2015/12/infamous_fx.jpg)
https://knarkowicz.wordpress.com/2016/01/09/automatic-exposure/

Local lights are not quite solved in games, and do not work with single-float exposure. Afaik GoW and Ghost of Tsushima games are using some sort of 2D/3D LUT for exposure, yet not sure how to implement it on top of existing content.

Current workaround for magic/light - bypass exposure completely. 

For local light:
Fillament falloff(physically accurate):
![изображение](https://github.com/Try/OpenGothic/assets/122894/8b3d9929-1a5b-4107-ae9b-7d4ae6a10ef4)
 
OpenGothic:
![изображение](https://github.com/Try/OpenGothic/assets/122894/31e341a8-e191-4d58-b133-0be52adc5978)

Fillament is too correct here, showcasing content problems, such as light from nowhere, so I've bump-down curve a little.

**Reply by madig** [reply]
Interesting stuff :) Are you interested in improved rendering of the old DX7 look, too? Both what I see in the DX11 mod and OpenGothic looks like an enthusiast's attempt at modern lighting techniques, but I found the game to look more cozy in its original low-fidelity rendering. Physically accurate lighting and the retro-painted look of Gothic's assets somehow don't mix well IMHO. If only I could see it in all the darkness on the screen :grimacing:

**Reply by Try** [reply]
> Are you interested in improved rendering of the old DX7 look, too?

Maintaining 2 different render-types is not an option
And what "improved" actually mean? Is per-pixel light an improval? Are shadows improval? RayTraced shadows?)

**Reply by madig** [reply]
Understood, not trying to make you do things you're not interested in :)

Well, I'd take the old look plus the much better rendering distance of modern games :)

**Reply by Try** [reply]
> If only I could see it in all the darkness on the screen 😬

This is quite subjective, to each eye/monitor, so gamma slider is necessity. One other interesting take on this is HDR-swapchain - allow engine to submit HDR image to system as-is and let the system care about gamma/tonemapping

**Comment by Roever1** [comment]
> Hi! I stumbled over this project recently, exciting.
> 
> I'm currently playing Chronicles of Myrtana and re-discovered how dark Gothic is, as in, I have to turn up the gamma because I can't see in broad daylight. In OpenGothic, the gamma sliders don't work...
> 
Hi 
are u saying that you run Chronicles of Myrtana with this engine? Does that work? Sorry if my question is off/stupid, im still trying to get OpenGothic to run in general in Linux :-D

**Reply by madig** [reply]
Oh, sorry, I didn't test that. I'm playing the game in the original engine, but the original, the DX11 mod and OpenGothic all share the can't-see-shit.

**Reply by Roever1** [reply]
okay thanks for the fast reply, i was very hyped for a scond :D

**Reply by Try** [reply]
![изображение](https://github.com/Try/OpenGothic/assets/122894/85d96b90-2d6b-466e-9388-9b18a6d89942)

You can launch CoM, but not a lot more, as they do require DMA and custom *.dll files

**Reply by Roever1** [reply]
me again ;)
When I try so start CoM I get the following error:
```
/OpenGothic/build/opengothic$ ./Gothic2Notr.sh -g /home/b/Games/"The Chronicles Of Myrtana Archolos"
OpenGothic v1.0 dev
invalid gothic path: "/home/b/Games/The Chronicles Of Myrtana Archolos/"

---crashlog(std::logic_error(gothic not found!))---
```
do i need to run a different script? For CoM start i also need to specify the Gothic install path, somehow this should be implement, i would assume?
Thanks again!

**Reply by Try** [reply]
Here is my commandline:
`-g "C:\Program Files (x86)\Steam\steamapps\common\TheChroniclesOfMyrtana" -game:TheChroniclesOfMyrtana.ini -rt 1 -nomenu -window`
